src/dotnet/HQ.Server/Services/TimeEntryServiceV1.cs

```diff
@@ -59,7 +59,13 @@ public async Task<Result<UpsertTimeV1.Response>> UpsertTimeV1(UpsertTimeV1.Reque
             }
 
 
-            var chargeCode = await _context.ChargeCodes.Where(t => t.Code == request.ChargeCode || t.Id == request.ChargeCodeId).FirstOrDefaultAsync();
+            var chargeCode = await _context.ChargeCodes.Where(t => t.Code == request.ChargeCode || t.Id == request.ChargeCodeId).FirstOrDefaultAsync(ct);
+            var maximumTimeEntryHours = await _context.Projects.Where(t => t.ChargeCode!.Id == request.ChargeCodeId).Select(t => t.TimeEntryMaxHours).SingleOrDefaultAsync(ct);
```

Comment:
Instead of reading the project separately, add `.Include(t => t.Project)` to the charge code and then check `chargeCode.Project.TimeEntryMaxHours`. - One less query to run.